server/handler/base.go

```diff
@@ -111,6 +122,87 @@ func (b *Base) postGitHubRepoStatus(ctx context.Context, client *github.Client,
 	return err
 }
 
+func (b *Base) PostCheckResults(ctx context.Context, client *github.Client, pr *github.PullRequest, evaluationResult common.Result) error {
+	sha := pr.GetHead().GetSHA()
+	owner := pr.GetBase().GetRepo().GetOwner().GetLogin()
+	repo := pr.GetBase().GetRepo().GetName()
+
+	for _, result := range evaluationResult.Children {
+		if err := b.createGitHubRepoCheck(ctx, client, pr, owner, repo, sha, *result); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *Base) createGitHubRepoCheck(ctx context.Context, client *github.Client, pr *github.PullRequest, owner string, repo string, sha string, evaluationResult common.Result) error {
+	publicURL := strings.TrimSuffix(b.BaseConfig.PublicURL, "/")
+	detailsURL := fmt.Sprintf("%s/details/%s/%s/%d", publicURL, owner, repo, pr.GetNumber())
+	defaultStatus := "completed"
+	time := github.Timestamp{
+		Time: time.Now(),
+	}
+
+	conclusion := "neutral"
+	switch evaluationResult.Status.String() {
+	case "approved":
+		conclusion = "success"
+	case "skipped":
+		conclusion = "neutral"
+	case "disapproved":
+		conclusion = "failure"
+	case "unknown":
+		conclusion = "failure"
+	case "pending":
+		conclusion = "failure"
```

Comment:
I've changed `pending` to have a `action_required` conclusion. This better follows the checks workflow whereby a check that has been run must have a completed status when it is not in progress. It also makes sure that an "update" isn't needed on a check, thus ensuring we don't (currently) need to store state in the bot.

Below is the workflow suggested by [GitHub](https://developer.github.com/v3/checks/runs/)
![](https://developer.github.com/assets/images/check_runs.png)

Reply:
Just to add, this does now mean there is no manual way to force the checks to re-run, it will always require an update to the rest of the PR. 

In the example below, the first check run has a status of `action_required` and the second has one of `failure`

![image](https://user-images.githubusercontent.com/380725/50839865-2aa53d00-1359-11e9-99ee-c40adfcbca75.png)